src/paperoni/utils.py

```diff
@@ -232,7 +232,9 @@ def __enter__(self):
         pass
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if exc_type is KeyboardInterrupt:
+        # Don't suppress KeyboardInterrupt, GeneratorExit or SystemExit - these
+        # must propagate
+        if exc_type in (KeyboardInterrupt, GeneratorExit, SystemExit):
```

Comment:
I think `GeneratorExit` and `SystemExit` should not be suppressed but you know better the inners of python and I can revert the changes if needed